types/artifacts.d.ts

```diff
@@ -551,6 +552,50 @@ declare global {
         /** The package name on NPM, if it exists. */
         npm?: string;
       }
+
+      export interface TimingSummary {
+        firstContentfulPaint: number | undefined;
+        firstContentfulPaintTs: number | undefined;
+        firstMeaningfulPaint: number | undefined;
+        firstMeaningfulPaintTs: number | undefined;
+        largestContentfulPaint: number | undefined;
+        largestContentfulPaintTs: number | undefined;
+        firstCPUIdle: number | undefined;
+        firstCPUIdleTs: number | undefined;
+        interactive: number | undefined;
+        interactiveTs: number | undefined;
+        speedIndex: number | undefined;
+        speedIndexTs: number | undefined;
+        estimatedInputLatency: number | undefined;
+        estimatedInputLatencyTs: number | undefined;
+        totalBlockingTime: number | undefined;
+        observedNavigationStart: number | undefined;
+        observedNavigationStartTs: number | undefined;
+        observedFirstPaint: number | undefined;
+        observedFirstPaintTs: number | undefined;
+        observedFirstContentfulPaint: number | undefined;
+        observedFirstContentfulPaintTs: number | undefined;
+        observedFirstMeaningfulPaint: number | undefined;
+        observedFirstMeaningfulPaintTs: number | undefined;
+        observedLargestContentfulPaint: number | undefined;
+        observedLargestContentfulPaintTs: number | undefined;
+        observedTraceEnd: number | undefined;
+        observedTraceEndTs: number | undefined;
+        observedLoad: number | undefined;
+        observedLoadTs: number | undefined;
+        observedDomContentLoaded: number | undefined;
+        observedDomContentLoadedTs: number | undefined;
+        observedFirstVisualChange: number | undefined;
+        observedFirstVisualChangeTs: number | undefined;
+        observedLastVisualChange: number | undefined;
+        observedLastVisualChangeTs: number | undefined;
+        observedSpeedIndex: number | undefined;
+        observedSpeedIndexTs: number | undefined;
+
+        debugInfo: Array<Record<string, unknown>>;
+
+        [p: string]: number | undefined | Array<Record<string, unknown>>;
```

Comment:
is this what you were referring to @connorjclark with ts gremlins when you wanted to add `lcpInvalidated` to metric object?

it'd be nice to avoid this @khempenius maybe we can just nest the artifact like `{metrics, debugInfo}` to keep consumers of just the metrics object happy with the `number|undefined` type?